### ISSUE TITLE
L051 enhancement: Replace implicit joins with INNER JOIN.

### ISSUE THREAD

**Issue by jpy-git** [sqlfluff/sqlfluff#2429]
This is another anti-pattern that should be fixed by L051.

This is an older SQL syntax (that's deprecated in a couple dialects now) for INNER JOIN:
```sql
SELECT *
FROM  a, b
WHERE a.id = b.id;
```

L051 should replace this with:
```sql
SELECT *
FROM  a
INNER JOIN b
    ON a.id = b.id;
```

**Comment by jpy-git** [comment]
Bit of complexity in that this is considered a CROSS JOIN:
```
SELECT * FROM a, b;
```

![image](https://user-images.githubusercontent.com/80432516/150654719-e8ac344c-497d-47e7-961c-b46c2efae73a.png)

So will have to be it's own rule rather than an extension of L051

**Comment by jivanpal** [comment]
Agreed that it should be flagged (except in MySQL, see below), but it's not necessarily a fixable rule. Consider (1):

```sql
SELECT *
FROM a
INNER JOIN b
    ON b.id_a = a.id
WHERE a.value = b.value
```

The user could equivalently write (2):

```sql
SELECT *
FROM a
INNER JOIN b
    ON b.id_a = a.id
    AND a.value = b.value
```

Or they could write (3):

```sql
SELECT *
FROM a, b
WHERE
    b.id_a = a.id
    AND a.value = b.value
```

Assuming `a.id` is a primary key and `b.id_a` is a foreign key referencing `a.id`, I would argue that the first of these queries be the canonical form, but without this knowledge, I don't think there is any good way to decide which form to choose. Even if we only flag/detect case (3), determining which of (1) or (2) it should be rewritten as is troublesome. Perhaps `b.id_a` isn't even a foreign key, and thus the user is really trying to do a CROSS JOIN.

Also note that MySQL/MariaDB behaviour of `CROSS JOIN` is different (there, it is synonymous with `[INNER] JOIN`); one *must* use `SELECT * FROM a, b` instead to perform what would be a CROSS JOIN in standard/ANSI SQL so it is not necessarily even an error/mistake in that dialect to do implicit joins — it definitely isn't when there is no WHERE clause.
